File: OJ practices/LeetCode题目集/27. Remove Element(easy).java
/**
 * @Date    : 2019-01-15 20:15:20
 * @Author  : 酸饺子 (changzheng300@foxmail.com)
 * @Link    : https://github.com/SourDumplings
 * @Version : $Id$
 *
 * https://leetcode.com/problems/remove-element/submissions/
*/

class Solution
{
    public int removeElement(int[] nums, int val)
    {
        if (nums.length == 0)
        {
            return 0;
        }
        int back = nums.length - 1;
        for (int i = 0; i <= back;)
        {
            if (nums[i] == val)
            {
                nums[i] = nums[back];
                --back;
            }
            else
                ++i;
        }
        return back + 1;
    }
}

File: OJ practices/LeetCode题目集/27. Remove Element(easy).cpp
/**
 * @brief 移除元素，双指针法
 * 
 */

/*
https://leetcode.com/problems/remove-element/
https://leetcode.cn/problems/remove-element/solutions/730203/yi-chu-yuan-su-by-leetcode-solution-svxi/?envType=study-plan-v2&envId=top-interview-150
 */

class Solution
{
public:
    int removeElement(vector<int> &nums, int val)
    {
        int n = nums.size();
        int l = 0;
        for (int r = 0; r < n; ++r)
        {
            if (nums[r] != val)
            {
                nums[l++] = nums[r];
            }
        }
        return l;
    }
};

